src/k8s-engine/k3sHelper.ts

```diff
@@ -199,12 +210,45 @@ export default class K3sHelper extends events.EventEmitter {
 
   protected async updateCache(): Promise<void> {
     try {
+      const recommended: Record<string, string[]> = {};
       let wantMoreVersions = true;
       let url = this.releaseApiUrl;
 
       await this.readCache();
 
-      console.log('Updating release version cache');
+      console.log(`Updating release version cache with ${ Object.keys(this.versions).length } items in cache`);
+      const channelResponse = await fetch(this.channelApiUrl, { headers: { Accept: this.channelApiAccept } });
+
+      if (channelResponse.ok) {
+        const channels = (await channelResponse.json()) as { data?: { name: string, latest: string }[] };
+        const nameSet: Record<string, string[]> = {};
+
+        console.log(`Got K3s update channel data: ${ channels.data?.map(ch => ch.name) }`);
+        for (const channel of channels.data ?? []) {
+          nameSet[channel.latest] = (nameSet[channel.latest] ?? []).concat(channel.name);
+        }
+        for (const [key, names] of Object.entries(nameSet)) {
+          recommended[key] = names.sort((a, b) => {
+            // The names are either a word ("stable", "testing") or a branch ("v1.2").
+            // The sort should be words first, then branch.
+            // For words, list "stable" before anything else.
+            // We assume no release can match two branch channels at once.
+            const versionRegex = /^v(?<major>\d+)\.(?<minor>\d+)$/;
+
+            if (a === 'stable' || b === 'stable') {
+              // sort "stable" at the front
+              return a === 'stable' ? -1 : 1;
+            }
+            if (versionRegex.test(a) || versionRegex.test(b)) {
+              return versionRegex.test(a) ? 1 : -1;
+            }
+
+            return a.localeCompare(b);
```

Comment:
According to the comment at line 232, all the names here must be `"testing"`, in which case you can return 0. Or should the comment be `a word ("stable", "testing", etc.)`?

Reply:
Fixed in 5a72fb1325fb57c624824a1f24f9624dd867b0b7 (the second option).